Shared/AuthenticationKeychainInfo.swift

```diff
@@ -63,6 +67,8 @@ open class AuthenticationKeychainInfo: NSObject, NSCoding {
         self.useTouchID = aDecoder.decodeAsBool(forKey: "useTouchID")
         if aDecoder.containsValue(forKey: "isPasscodeRequiredImmediately") {
             self.isPasscodeRequiredImmediately = aDecoder.decodeAsBool(forKey: "isPasscodeRequiredImmediately")
+        } else if let interval = aDecoder.decodeObject(forKey: "requiredPasscodeInterval") as? NSNumber {
+            self.isPasscodeRequiredImmediately = (interval == 0)
```

Comment:
`interval` should never be 0 based on 1.6 since `requiredPasscodeInterval` is an `Int` enum which starts at `immediate = 2` and [these lines](https://github.com/brave/browser-ios/blob/development/Shared/AuthenticationKeychainInfo.swift#L81): 

```swift
if var interval = aDecoder.decodeObject(forKey: "requiredPasscodeInterval") as? NSNumber {
    // We have updated the immediate lockout value to 2 from 0 due to timing issues with systemUptime()
    interval = interval == 0 ? 2 : interval
    self.requiredPasscodeInterval = PasscodeInterval(rawValue: interval.intValue)
}
```

So probably best to make this `self.isPasscodeRequiredImmediately = (interval == 2)` or even `(interval <= 2)` for safety

Reply:
I think the check is for 0 which if tru converts it into enum value of 2. else the fallback.